### PR TITLE
Expose user permissions

### PR DESCRIPTION
**Pull request by kenoir**
Expose permissions for current user in order to provide this info to the UI client so controls can be varied according to permissions.

This is specifically to add controls around paid imagery.

Example response:

`GET /permissions`

```json
{
  "data": {
    "description": "Available permissions for current user (you)",
    "permissionsSet": {
      "user": {
        "email": "nope@guardian.co.uk",
        "firstName": "Robert",
        "lastName": "Kenny",
        "avatarUrl": "https://lh3.googleusercontent.com/nope/photo.jpg"
      },
      "permissions": [
        "editMetadata",
        "deleteImage",
        "deleteCrops",
        "bigSpender"
      ],
      "lastUpdated": 1458298190366
    }
  }
}
```

### CODE REVIEW

File: media-api/app/controllers/PermissionsController.scala [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+object PermissionsController extends Controller with ArgoHelpers {
+
+  val Authenticated = Authed.action
+  val permissionStore = Authed.permissionStore
+
+  def permissionsResponse(permissions: Set[PermissionType.PermissionType]) = {
+    val perms = Json.toJson(permissions.map(_.toString))
+
+    val permsData = Json.obj(
+      "description" -> "Available permissions for current user (you)",
```
Comment (akash1810) [inline media-api/app/controllers/PermissionsController.scala]: worth adding the current username/email here?

File: common-lib/src/main/scala/com/gu/mediaservice/lib/auth/Store.scala [hunk 2]
```diff
@@ -68,9 +68,20 @@ object PermissionType extends Enumeration {
   val EditMetadata = Value("editMetadata")
   val DeleteImage  = Value("deleteImage")
   val DeleteCrops  = Value("deleteCrops")
+  val BigSpender   = Value("bigSpender")
```
Comment (akash1810) [inline common-lib/src/main/scala/com/gu/mediaservice/lib/auth/Store.scala]: might want to add this key to the [sample permissions file](https://github.com/guardian/grid/blob/master/cloud-formation/scripts/permissions.properties) too

**Comment by akash1810** [comment]
The `permissionStore` is updated on an Akka schedule, would it be worth adding a `lastUpdated` property?

**Comment by kenoir** [comment]
Closing for now, as we're unsure how helpful this will be.

**Comment by NickPapacostas** [comment]
:+1:

**Pull request merged**